apps/nycbus/nyc_bus.star

```diff
@@ -0,0 +1,159 @@
+"""
+Applet: NYC Bus
+Summary: NYC Bus departures
+Description: Real-time bus departures for your preferred stop.
+Author: samandmoore
+"""
+
+load("cache.star", "cache")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+load("time.star", "time")
+
+API_KEY = "<INSERT_API_KEY_HERE>"
+STOP_CODE = "550685"
+BUSTIME_STOP_TIMES_URL = "http://bustime.mta.info/api/siri/stop-monitoring.json"
+BUSTIME_STOP_INFO_URL = "http://bustime.mta.info/api/where/stop/%s.json"
+
+def get_schema():
+    stop_options = [
+        schema.Option(
+            display = "21 ST/BROADWAY SW",
+            value = "550685",
+        ),
+        schema.Option(
+            display = "21 ST/41 AV NE",
+            value = "550670",
+        ),
+    ]
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Location(
+                id = "location",
+                name = "Location",
+                desc = "Location to use for nearby stops.",
+                icon = "place",
+            ),
+            schema.Dropdown(
+                id = "stop_code",
+                name = "Stop Code",
+                desc = "The stop code to use.",
+                icon = "bus",
+                default = stop_options[1].value,
+                options = stop_options,
+            ),
+        ],
+    )
+
+def main(config):
+    api_key = config.get("api_key") or API_KEY
+    stop_code = config.get("stop_code") or STOP_CODE
+    journeys = get_journeys(api_key, stop_code)
+    first_journey = journeys[0]
+    second_journey = journeys[1]
+
+    return render.Root(
+        child=render.Column(
+            expanded=True,
+            main_align="space_evenly",
+            children=[
+                build_row(first_journey),
+                render.Box(
+                    width=64,
+                    height=1,
+                    color="#f5f5f5",
+                ),
+                build_row(second_journey),
+            ]
+        )
+    )
+
+def build_row(journey):
+    return render.Row(
+        expanded=True,
+        main_align="space_evenly",
+        children=[
+            render.Circle(
+                color="#%s" % journey["line_color"],
+                diameter=14,
+                child=render.Text(journey["line_name"], color="#fff", font="CG-pixel-3x5-mono"),
```

Comment:
this is the smallest font I could find. It works great for bus lines with 2-3 characters (M1, Q69, etc) but it looks less awesome for 4 character bus lines (Q100).

Reply:
It's the smallest font we can find as well 😅 . I'm not sure it gets any smaller - it's a pretty impressively small-yet-readable font.